pubtools/_quay/push_docker.py

```diff
@@ -731,7 +727,28 @@ def run(self) -> None:
                 failed = True
 
         # Remove old signatures
-        outdated_manifests = self.get_outdated_manifests(backup_tags)
+        # run generate backup mapping again to fetch new digests of pushed containers
+        backup_tags2, _ = self.generate_backup_mapping(docker_push_items)
+        # if new backup tag has differnet digest, it means it was overwritten during the push
+        # and old signature should be removed. If the digest is the same it means, same item
+        # was just repushed
+        outdated_tags = {}
+        backup_tags2_shared = {}
+
+        # Backup tags can contain new tags which were orignally rollback_tags
+        # limit the comparision for outdated manifests to original backup_tags only
+        for bt2 in backup_tags2.items():
+            if bt2[0] in backup_tags:
+                backup_tags2_shared[bt2[0]] = bt2[1]
```

Comment:
bt2[0] is ImageData(repo=repo1, tag=tag1, v2list_digest=digest1, v2s2_digest=digest2, v2s1_digest=digest3). If digest has changed for a tag, bt2[0] will be a different ImageData and won't exist in backup_tags.

Reply:
backup_tags2_shared doesn't includes changed digests due to that.

Reply:
yeah, good point